### PR TITLE
Un-invert `Options.disableDespawning` data byte

### PR DESCRIPTION
**Pull request by aubergine10**
For some reason the value of `Options.disableDespawning` was historically inverted during de/serialization.

```csharp
GameplayTab_VehicleBehaviourGroup.DisableDespawning.Value = !LoadBool(data, idx: 15, true); // inverted
```

This PR fixes that by making the serialized value conistent with how all other options work (not inverted), whilst handling legacy value (inverted) if necessary.

It's based in part on new `Version` field that @kianzarrin put in PR #1211 (note: if merging prior to that PR, Kian would need to change version in his PR to `2`).

Also fixes some trivial analyzer warnings I noticed.

### CODE REVIEW

**Comment by kianzarrin** [comment]
Damn my LaneConnectionManager PR  also adds version! this is going to conflict.

**Review by kvakvs — APPROVED** [review]
A risky endeavour, but alright, good intention

File: TLM/TLM/Manager/Impl/OptionsManager.cs [hunk 1]
```diff
@@ -135,8 +138,14 @@ private static void ToSlider(byte[] data, uint idx, ILegacySerializableOption op
                 ToCheckbox(data, idx: 11, OverlaysTab_OverlaysGroup.VehicleRestrictionsOverlay, false);
                 ToCheckbox(data, idx: 12, GameplayTab_VehicleBehaviourGroup.StrongerRoadConditionEffects, false);
                 ToCheckbox(data, idx: 13, PoliciesTab_AtJunctionsGroup.AllowUTurns, false);
-                ToCheckbox(data, idx: 14, PoliciesTab_AtJunctionsGroup.AllowLaneChangesWhileGoingStraight, false);
-                GameplayTab_VehicleBehaviourGroup.DisableDespawning.Value = !LoadBool(data, idx: 15, true); // inverted
+                ToCheckbox(data, idx: 14, PoliciesTab_AtJunctionsGroup.AllowLaneChangesWhileGoingStraight, true);
```
Comment (krzychu124) [inline TLM/TLM/Manager/Impl/OptionsManager.cs:141]: `AllowLaneChangesWhileGoingStraight` default value changed to true?

**Review by krzychu124 — CHANGES_REQUESTED** [review]

File: TLM/TLM/Manager/Impl/OptionsManager.cs (hunk 1 above)
Comment (aubergine10) [reply TLM/TLM/Manager/Impl/OptionsManager.cs:141]: Ah, I blundered fixing the merge conflict with https://github.com/CitiesSkylinesMods/TMPE/pull/1455/commits/0d2f593d90ee882305a36ff9fb9111fa757d755a

**Review by krzychu124 — APPROVED** [review]
ok 👍

**Comment by krzychu124** [comment]
I'm not sure why I didn't notice it but.... if we are starting "new game", there won't be Configuration data so `public static int Version => _configuration.Version;` will throw null ref exception

**Pull request merged**